python/lsst/ap/verify/ingestion.py

```diff
@@ -25,31 +25,343 @@
+
+        Parameters
+        ----------
+        repo : `str`
+            The output repository location on disk for raw images. Must exist.
+        dataFiles : `list` of `str`
+            A list of filenames to ingest. May contain wildcards.
+        badFiles : `list` of `str`
+            A list of filenames to exclude from ingestion. Must not contain paths.
+            May contain wildcards.
+        """
+        args = [repo, "--filetype", "raw", "--mode", "link"]
+        if badFiles:
+            args.append('--badFile')
+            args.extend(badFiles)
+        args.extend(dataFiles)
+        try:
+            _runIngestTask(self.dataIngester, args)
+        except sqlite3.IntegrityError as detail:
+            raise_from(RuntimeError("Not all raw files are unique"), detail)
+
+    def _ingestCalibs(self, dataset, workspace):
+        """Ingest the calibration files for use by LSST.
+
+        After this method returns, the calibration repository in ``workspace``
+        shall contain all calibration data from ``dataset``. Butler operations
+        on the repository shall not be able to modify ``dataset``.
+
+        Parameters
+        ----------
+        dataset : `lsst.ap.verify.dataset.Dataset`
+            The dataset on which the pipeline will be run.
+        workspace : `lsst.ap.verify.workspace.Workspace`
+            The location containing all ingestion repositories.
+        """
+        if os.path.exists(os.path.join(workspace.calibRepo, "calibRegistry.sqlite3")):
+            self.log.info("Calibration files were previously ingested, skipping...")
+        else:
+            self.log.info("Ingesting calibration files...")
+            calibDataFiles = _findMatchingFiles(dataset.calibLocation,
+                                                self.config.calibFiles, self.config.calibBadFiles)
+            self._doIngestCalibs(workspace.dataRepo, workspace.calibRepo, calibDataFiles)
+            self.log.info("Calibrations corresponding to {0} are now ingested in {1}".format(
+                workspace.dataRepo, workspace.calibRepo))
+
+    def _doIngestCalibs(self, repo, calibRepo, calibDataFiles):
+        """Ingest callibration images into a calibration repository.
+
+        Parameters
+        ----------
+        repo : `str`
+            The output repository location on disk for raw images. Must exist.
+        calibRepo : `str`
+            The output repository location on disk for calibration files. Must
+            exist.
+        calibDataFiles : `list` of `str`
+            A list of filenames to ingest. Supported files vary by instrument
+            but may include flats, biases, darks, fringes, or sky. May contain
+            wildcards.
+        """
+        # TODO: --output is workaround for DM-11668
+        args = [repo, "--calib", calibRepo, "--output", os.path.join(calibRepo, "dummy"),
+                "--mode", "link", "--validity", "999"]
+        args.extend(calibDataFiles)
```

Comment:
Something else? I got exactly the same error as Colin describes on DM-11668, and it started after I let the raw repo be new-style.

I just copied `--validity` from your code, so if you think it should be configurable, I'll do that.

Reply:
"Something else" as in "wow." Must be a new-style butler thing, yes. Sorry for the noise there.

Since so many other things are configurable now, it seems reasonable to not have validity hard-coded either. But a default value of 999 is fine. This may well be another DECam-ism; in other words, other cameras may rely on the validity parameter to set registry values during ingestion more than DECam typically does.